gum/backend-x86/gumstalker-x86.c

```diff
@@ -2080,6 +2182,25 @@ gum_exec_ctx_free (GumExecCtx * ctx)
   gum_x86_writer_clear (&ctx->slow_writer);
   gum_x86_writer_clear (&ctx->code_writer);
 
+#ifdef HAVE_LINUX
+  {
+    GumStalker * stalker = ctx->stalker;
+    GumInterceptor * interceptor;
+
+    interceptor = g_steal_pointer (&stalker->interceptor);
+    if (interceptor != NULL)
+    {
+      if (_Unwind_Find_FDE_impl != NULL)
+        gum_interceptor_revert (interceptor, _Unwind_Find_FDE_impl);
+
+      if (__gxx_personality_v0_impl != NULL)
+        gum_interceptor_revert (interceptor, __gxx_personality_v0_impl);
```

Comment:
Can you point me at a suitable example where this is done so I can copy the approach?

Reply:
I'd say `bindings/gumjs/gumscriptbackend.c`.

Reply:
I've got a bit of a problem with this. The hook functions need to be able to access the GMetalHashTable containing the lookup information. The only way for these functions to retrieve this information is by gaining a handle to the `GumStalker`. Thus the hooks cannot outlive the Stalker instance itself. I shall move the instantiation accordingly.

Reply:
But they can retrieve the `ExecCtx` through the thread-local data though?

Reply:
Oh, that's currently on the `Stalker` instance. I think we should make it global, like we do for `gum_interceptor_context_private`.

Reply:
I’m not sure I follow? Also if the hash table isn’t attached to the context, then if we don’t purge the related entries from the hashtable, if the same memory pages are reused for another instance slab,  we can end up with collisions?

Reply:
I mean the hash table should be attached to the context, like it is now. But the thread-local key used to find the context, shouldn't be attached to the `Stalker` instance. This is because it doesn't make sense for two different `Stalker` instances to stalk the same thread at the same time.